src/apiRequest/ApiRequestReducer.js

```diff
@@ -0,0 +1,43 @@
+import {
+  API_REQUEST,
+  API_REQUEST_SUCCESS,
+  API_REQUEST_FAILURE,
+  CLEAR_API_RESPONSE,
+} from './ApiRequestActionTypes';
+
+export default function apiRequestReducer(
+  state = {
+    isLoading: false,
+    response: undefined,
+    error: undefined,
+  }, action) {
+  const {data, error} = action;
+
+  switch (action.type) {
+    case API_REQUEST:
+      return {
+        ...state,
+        isLoading: true,
+      };
+    case API_REQUEST_SUCCESS:
+      return {
+        ...state,
+        isLoading: false,
+        response: data,
+      };
+    case API_REQUEST_FAILURE:
+      return {
+        ...state,
+        isLoading: false,
+        error,
+      };
+    case CLEAR_API_RESPONSE:
+      return {
+        ...state,
+        response: undefined,
+        error: undefined,
```

Comment:
I think that `isLoading` also should be explicitly cleared. :slightly_smiling_face: 
We assume that `CLEAR` actions reset everything.